tests/functional/test_basic.py

```diff
@@ -195,7 +189,9 @@ def test_python_makefile(pyi_builder):
 
 
 def test_set_icon(pyi_builder, data_dir):
-    icon_dir = os.path.join(data_dir, 'icons')
+    # Note that the name test_set_icon (not just test_icon) for prevents
+    # data_dir from copying data/icon to the tmpdir.
+    icon_dir = os.path.join(data_dir.strpath, 'icons')
```

Comment:
Note that as written, `strpath` is the original _DATA_DIR folder, and not the test-specific folder that was copied to the tmpdir. I would have simplified everything by using only the copy in the tmpdir, and used `data_dir` as a `py.path.local` to do path manipulation.